api/tests/unit/domain/validators/schooling-registration-validator_test.js

```diff
@@ -142,41 +166,73 @@ describe('Unit | Domain | Schooling Registration validator', () => {
+            }  
+          });
+    
+          it('respects INSEE Code, only number', async () => {
+            try {
+              checkValidation({ ...validAttributes, birthCountryCode: FRANCE_COUNTRY_CODE, birthCityCode: '2B125' });
+            } catch (e) {
+              expect.fail('SchoolingRegistration is valid birthCityCode respect INSEE code, like Corsica');
+            }  
+          });    
+        });
 
-        try {
-          checkValidation({ ...validAttributes, birthCountryCode: '12345', birthCityCode: undefined });
-        } catch (e) {
-          expect.fail('SchoolingRegistration is valid when birthCity is undefined and birthCountry is not France');
-        }
+        context('when birthCountryCode not equal to France', () => {
+          it('is valid with birthCityCode undefined', async () => {
+            try {
+              checkValidation({ ...validAttributes, birthCountryCode: '99125', birthCityCode: undefined });
+            } catch (e) {
+              expect.fail('SchoolingRegistration is valid when birthCity is undefined and birthCountry is not France');
+            }
+          });
+        });
       });
 
-      it('throw an error when birthCityCode has not 5 characters', async () => {
-        const error = await catchErr(checkValidation)({ ...validAttributes, birthCityCode: '1' });
-
-        expect(error.key).to.equal('birthCityCode');
-        expect(error.why).to.equal('length');
-        expect(error.limit).to.equal(5);
+      context('is invalid', () => {
+        context('when birthCountryCode is equal to France', () => {
+          it('throw an error with a birthCityCode of 5 characters', async () => {
+            const error = await catchErr(checkValidation)({ ...validAttributes, birthCountryCode: FRANCE_COUNTRY_CODE, birthCityCode: '123456' });
+    
+            expect(error.key).to.equal('birthCityCode');
+            expect(error.why).to.equal('length');
+            expect(error.limit).to.equal(5);
+          });
+    
+          it('throws an error with a birthCityCode which has a letter not in second character', async () => {
+            const error = await catchErr(checkValidation)({ ...validAttributes, birthCountryCode: FRANCE_COUNTRY_CODE, birthCityCode: '21B22' });
+    
+            expect(error.key).to.equal('birthCityCode');
+            expect(error.why).to.equal('not_match_regex');
+          });
+            
+          context('when birthCountryCode is not equal to France', () => {
+            it('throws an error with a birthCityCode of not 5 characters', async () => {
+              const error = await catchErr(checkValidation)({ ...validAttributes, birthCityCode: '1' });
+    
+              expect(error.key).to.equal('birthCityCode');
+              expect(error.why).to.equal('length');
+              expect(error.limit).to.equal(5);
+            });
+          });
```

Comment:
Le contexte `when birthCountryCode is equal to France` doit se fermer avant l'ouverture du contexte `when birthCountryCode is not equal to France `
```suggestion
          });
      });
          
      context('when birthCountryCode is not equal to France', () => {
          it('throws an error with a birthCityCode of not 5 characters', async () => {
          const error = await catchErr(checkValidation)({ ...validAttributes, birthCityCode: '1' });
          expect(error.key).to.equal('birthCityCode');
          expect(error.why).to.equal('length');
          expect(error.limit).to.equal(5);
        });
```